superset/assets/src/visualizations/leaflet_map.js

```diff
@@ -372,6 +373,54 @@ function leafletmap(slice, payload) {
         renderBasicMap();
         renderPolygonLayer();
         addColumnsDropdownToMap();
+        addHeatMapColorLegend();
+    }
+
+    function addHeatMapColorLegend(){
+      var colname  = getSelectedColorColumn()
+      var col = colorCols[colname];
+      var minValue = col['operator'];
+      var maxvalue = col['sqlExpression'];
+      var legends = splitRangeIntoEqualParts(parseFloat(minValue), parseFloat(maxvalue), 10);
+
+      var legend = L.control({ position: 'topleft' });
+      legend.onAdd = () => {
+        var div
+        if($('#heat-map-color-legend').length > 0){
+          div = $('#heat-map-color-legend')[0];
+        } else {
+          div = L.DomUtil.create('div');
+          div.setAttribute('id', 'heat-map-color-legend')
+
```

Comment:
it is neccessary to set this fix id

Reply:
yes, because using this id checking that already legend component exist or not.. if not than creating new otherwise replacing values